File: frontend/src/types/video_gen.ts
/** 视频生成多模型相关类型（与后端 schemas/video_gen.py 对应）。 */

export type VideoGenProviderType = 'cogvideox' | 'kling' | 'minimax';

export interface VideoGenProviderMeta {
  provider_type: VideoGenProviderType;
  label: string;
  icon_key: string;
  default_base_url: string;
  default_model: string;
  supports_thinking: boolean;
  models: { model_id: string; label: string }[];
  key_label: string;
  key_url: string;
}

export interface VideoGenProviderConfig {
  id: string;
  provider_type: VideoGenProviderType;
  name: string;
  base_url: string;
  model_id: string;
  is_enabled: boolean;
  is_active: boolean;
  sort_order: number;
  api_key_masked: string;
  is_configured: boolean;
  created_at: string;
  updated_at?: string | null;
}

export interface VideoGenTestResult {
  ok: boolean;
  message: string;
}

export interface VideoGenConfigFormValues {
  provider_type: VideoGenProviderType;
  name: string;
  api_key: string;
  base_url: string;
  model_id: string;
  is_enabled: boolean;
  is_active: boolean;
  sort_order: number;
}

File: backend/app/schemas/video_gen.py
"""视频生成 Provider 配置 Schema。"""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class VideoGenProviderConfigCreate(BaseModel):
    """创建视频生成 provider 配置。"""

    provider_type: str = Field(..., description="cogvideox | kling | minimax")
    name: str = Field(..., min_length=1, max_length=100)
    api_key: str = Field(..., min_length=1, max_length=255)
    base_url: str = Field(default="", max_length=255)
    model_id: str = Field(default="", max_length=100)
    is_enabled: bool = True
    is_active: bool = False
    sort_order: int = 0


class VideoGenProviderConfigUpdate(BaseModel):
    """更新视频生成 provider 配置 — 全字段可选；api_key 留空表示不变。"""

    name: str | None = Field(default=None, max_length=100)
    api_key: str | None = Field(default=None, max_length=255)
    base_url: str | None = Field(default=None, max_length=255)
    model_id: str | None = Field(default=None, max_length=100)
    is_enabled: bool | None = None
    sort_order: int | None = None


class VideoGenProviderConfigResponse(BaseModel):
    """视频生成 provider 配置响应（api_key 脱敏）。"""

    model_config = {"from_attributes": True}

    id: UUID
    provider_type: str
    name: str
    base_url: str
    model_id: str
    is_enabled: bool
    is_active: bool
    sort_order: int
    api_key_masked: str = ""
    is_configured: bool = True
    created_at: datetime
    updated_at: datetime | None = None


class VideoGenTestResult(BaseModel):
    """视频生成 provider 连通性测试结果。"""

    ok: bool
    message: str
